migrations/upgrades/20191128055729_update_directus_files.php

```diff
@@ -0,0 +1,90 @@
+<?php
+
+
+use Phinx\Migration\AbstractMigration;
+use function \Directus\get_random_string;
+
+class UpdateDirectusFiles extends AbstractMigration
+{
+    public function change()
+    {
+        $fieldsTable = $this->table('directus_fields');
+        $filesTable = $this->table('directus_files');
+        
+        if($this->checkFieldExist('directus_files', 'filename')){
+            $this->execute(\Directus\phinx_update(
+                $this->getAdapter(),
+                'directus_fields',
+                [
+                    'field' => 'filename_disk'
+                ],
+                ['collection' => 'directus_files', 'field' => 'filename']
+            ));
+        }
+
+        if (!$filesTable->hasColumn('filename_disk')) {
+            $filesTable->renameColumn('filename', 'filename_disk');
+        }
+    
+        if (!$filesTable->hasColumn('filename_download')) {
+            $filesTable->addColumn('filename_download', 'string', [
+                'limit' => 255,
+                'null' => true,
+                'default' => null
+            ])->save();
+        }
+        if(!$this->checkFieldExist('directus_files', 'filename_download')){     
+            $fieldsTable->insert([
+                'collection' => 'directus_files',
+                'field' => 'filename_download',
+                'type' => 'string',
+                'interface'=>'text-input',
+                'locked' => 1,
+                'readonly' => 1,
+                'hidden_detail' => 1,
+                'hidden_browse' => 1
+            ])->save();
+        }
+        if($filesTable->hasColumn('id')){
+            $filesTable->changeColumn('id', 'integer');
+        }
+
+        if($this->checkFieldExist('directus_files','id')){
+            $this->execute(\Directus\phinx_update(
+                $this->getAdapter(),
+                'directus_fields',
+                [
+                    'type' => 'integer'
+                ],
+                ['collection' => 'directus_files', 'field' => 'id']
+            ));
+        }
+
+        // adds private_hash in directus_files collection for existing data
+        
+        if ($filesTable->hasColumn('private_hash')) {
```

Comment:
Does the private hash column exist at this point? I don't see any "add private_hash" column migration above this statement, so I'm thinking the v7 users won't get this.

Reply:
https://github.com/directus/api/blob/develop/migrations/upgrades/20191123053252_update_directus_files_field.php#L164